### DISCUSSION TITLE
Measure and Outline addons Turn On Automatically

### DISCUSSION THREAD

**Discussion by toddbrady** [storybookjs/storybook#15339] [Help]
After migrating to Storybook 6.3 from 6.2.8 there seems to be a persistent issue occurring with the new Layout Debugging features. 

It appears that when working in Storybook (at least locally for now), that the layout feature for the measurement tool turns on automatically. What's the best way to possibly track the cause of this? It seems that URL automatically gets appended as well with: `&globals=measureEnabled:true` without ever clicking or enabling the measure icon.

**Comment by winkerVSbecks** [comment]
@toddbrady I'm not able to reproduce this. What's framework are you working with, React or something else?

**Reply by toddbrady** [reply]
@winkerVSbecks Yes, it's Create React App on Windows. I think I know what the issue is...
`alt + tab` to switch apps on Windows. The `alt` key appears to be triggering the measure icon. I'm assuming this is the keyboard shortcut for it? Is there a way to change it? Should this be reported as a bug?

**Reply by toddbrady** [reply]
I attempted to modify `manager.js` with: 
```
addons.setConfig({
    enableShortcuts: false
});
```
That seemed to disable keyboard shortcuts (i.e. "D" for showing / hiding the addons panel). However, the `alt` key still seems to trigger the measurement tool.

**Reply by winkerVSbecks** [reply]
Tracking it here https://github.com/storybookjs/addon-measure/issues/15 Indeed alt+tab seems to be the culprit. Will work on a fix and the option to change the shortcut.

**Reply by winkerVSbecks** [reply]
@toddbrady could you test this out this canary version and see how it works for you `npm install @storybook/addon-measure@1.3.0--canary.17.2bc695e.0`

**Reply by toddbrady** [reply]
@winkerVSbecks yes, I will test tonight and report back

**Reply by winkerVSbecks** [reply]
Actually, I published it. So give this one a go: `npm install @storybook/addon-measure@1.3.0`

**Reply by toddbrady** [reply]
@winkerVSbecks 
I installed `@storybook/addon-measure@1.3.0`  in my `package.json` 

within `main.js` I have: 
```
addons: [
		'@storybook/addon-links',
		'@storybook/addon-essentials',
		'@storybook/preset-create-react-app',
		'storybook-addon-designs',
		'@etchteam/storybook-addon-status',
		'@storybook/addon-a11y'
	]
```

The behavior I see know still seems a little buggy. Some of the issues I'm seeing include: 
- clicking anywhere within the Storybook application automatically appends the URL with `&globals=measureEnabled:false` 
- if the URL receives the measure parameter, as soon as `alt` is pressed, the measure tool is activated. 
- also, other keyboard shortcuts such as "D" to toggle the drawer, have stopped working with this update. pressing other keyboard keys, automatically triggers the URL parameter for the measure tool.  

I'm not sure what the exact functionality should be. It seems like binding it to the `alt` key is still an issue and focus state within the application is causing it turn on automatically.

**Reply by winkerVSbecks** [reply]
Good catch on the other shortcuts not working. I patched that in `1.3.1`

The expected functionality is that you hold down `alt`/`⌥` to enable measure. When you release `alt` it disables the addon. The URL simply reflects the state of the addon. This is what it looks like:

![Kapture 2021-06-24 at 20 28 25](https://user-images.githubusercontent.com/42671/123351421-0e96b580-d52b-11eb-9505-eb3fdc022d1f.gif)

**Reply by toddbrady** [reply]
@winkerVSbecks this appears to be the expected behavior now on Windows. Other shortcut keys appear to be working as expected. The measure utility is only activated if there is a focus state on the main frame of Storybook when the "alt" key is activated.

**Reply by winkerVSbecks** [reply]
Thanks. I'm also very curious to know what happens with Figma. Do you get the same behaviour?

**Reply by toddbrady** [reply]
I don't see any unusual behavior with the Figma files that are linked in the Design tab. Though the Figma links we use for most of our components are fairly static and don't show any interactivity, just high level specs for what the final rendered component should look like.

**Reply by winkerVSbecks** [reply]
I meant on Figma itself. It uses alt to measure shortcut too. I was wondering if that has the same behaviour as Storybook.

**Reply by toddbrady** [reply]
Gotcha. I believe in Figma, and I could be mistaken, the "alt" key is used to measure distance between objects or layers when one object is selected and doesn't measure anything when there is not an active selection. Using the "alt" key doesn't actively provide details unless something is selected, and when an object is selected, details are shown in the "Design" panel.
